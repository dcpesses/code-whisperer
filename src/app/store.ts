import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import counterReducer from '@/features/redux-counter/counterSlice';
import modalCommandListReducer from '@/features/modal-command-list/modalSlice';
// import twitchReducer from '../features/twitch/twitchSlice';
import channelReducer from '@/features/twitch/channel-slice.js';
import userReducer from '@/features/player-queue/user-slice.js';
import queueReducer from '@/features/player-queue/queue-slice';
import todosReducer from '@/features/todos/todosSlice';

const reducer = {
  counter: counterReducer,
  channel: channelReducer,
  modal: modalCommandListReducer,
  queue: queueReducer,
  user: userReducer,
  todos: todosReducer,
};
export const store = configureStore({ reducer });

export function getStoreWithState(preloadedState?: RootState) {
  return configureStore({ reducer, preloadedState });
}

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
