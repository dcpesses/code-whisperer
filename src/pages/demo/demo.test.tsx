import {render, screen} from '@testing-library/react';
import {HashRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {store} from '@/app/store';

import Demo from './index';

describe('Demo', () => {
  test('Should render as expected', () => {
    const {container} = render(
      <Provider store={store}>
        <HashRouter>
          <Demo />
        </HashRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
  test('Should render with Home link', () => {
    render(
      <Provider store={store}>
        <HashRouter>
          <Demo />
        </HashRouter>
      </Provider>
    );

    expect(screen.getByText(/Home/i)).toBeInTheDocument();
  });
});
