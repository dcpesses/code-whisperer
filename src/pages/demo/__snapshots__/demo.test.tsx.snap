// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Demo > Should render as expected 1`] = `
<div>
  <div
    class="Demo"
  >
    <nav
      class="navbar navbar-expand navbar-light bg-dark sticky-top"
      data-bs-theme="dark"
    >
      <div
        class="container"
      >
        <span
          class="navbar-brand"
        >
          <a
            class="navbar-brand"
            href="#/"
          >
            ← Home
          </a>
        </span>
      </div>
    </nav>
    <div
      class="text-center pt-2 container"
    >
      <small>
        Oh, hey, you found some...
      </small>
      <h2
        class="py-3"
      >
        Demos & Examples
      </h2>
    </div>
    <div
      class="Demo-examples container"
      data-bs-theme="dark"
    >
      <div
        class="row"
      >
        <div
          class="Demo-example col-lg-4"
        >
          <div
            class="shadow h-100 card bg-dark text-white border-primary"
          >
            <div
              class="card-body"
            >
              <div
                class="pb-1 border-bottom border-secondary-subtle card-title h5"
              >
                Redux To-Dos
              </div>
              <div
                class="todos"
              >
                <div
                  class="add-todo"
                >
                  <input
                    placeholder="Enter a Todo..."
                    type="text"
                    value=""
                  />
                  <button>
                    Add Todo
                  </button>
                </div>
                <ul
                  class="todos-list"
                >
                  <li
                    class="todo-list-item"
                  >
                    <span
                      class="item-checkbox"
                    >
                      <input
                        title="Mark Complete"
                        type="checkbox"
                      />
                    </span>
                    <span
                      class="item-content"
                      style="text-decoration: none;"
                    >
                      Hit the gym
                    </span>
                    <span
                      class="item-actions"
                    >
                      <button
                        class="item-actions-edit"
                        title="Edit"
                      >
                        ✎
                      </button>
                      <button
                        class="item-actions-delete"
                        title="Delete"
                      >
                        ✖
                      </button>
                    </span>
                  </li>
                  <li
                    class="todo-list-item"
                  >
                    <span
                      class="item-checkbox"
                    >
                      <input
                        title="Mark Incomplete"
                        type="checkbox"
                      />
                    </span>
                    <span
                      class="item-content"
                      style="text-decoration: line-through;"
                    >
                      Meet George
                    </span>
                    <span
                      class="item-actions"
                    >
                      <button
                        class="item-actions-edit"
                        title="Edit"
                      >
                        ✎
                      </button>
                      <button
                        class="item-actions-delete"
                        title="Delete"
                      >
                        ✖
                      </button>
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Demo-example col-lg-4"
        >
          <div
            class="shadow h-100 card bg-dark text-white border-primary"
          >
            <div
              class="card-body"
            >
              <div
                class="pb-1 border-bottom border-secondary-subtle card-title h5"
              >
                Redux Counter
              </div>
              <div>
                <div
                  class="_row_503e3a"
                >
                  <button
                    aria-label="Decrement value"
                    class="_button_503e3a"
                  >
                     - 
                  </button>
                  <span
                    aria-label="Counter value"
                    class="_value_503e3a"
                  >
                    0
                  </span>
                  <button
                    aria-label="Increment value"
                    class="_button_503e3a"
                  >
                     + 
                  </button>
                </div>
                <div
                  class="_row_503e3a"
                >
                  <input
                    aria-label="Set increment amount"
                    class="_textbox_503e3a"
                    value="2"
                  />
                  <button
                    aria-label="Add increment amount"
                    class="_button_503e3a"
                  >
                    Add
                  </button>
                  <div>
                    <button
                      aria-label="Add async increment amount"
                      class="_asyncButton_503e3a"
                    >
                      Add Async
                    </button>
                    <button
                      aria-label="Add increment amount if odd"
                      class="_button_503e3a"
                    >
                      Add If Odd
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Demo-example col-lg-4"
        >
          <div
            class="shadow h-100 card bg-dark text-white border-primary"
          >
            <div
              class="card-body"
            >
              <div
                class="pb-1 border-bottom border-secondary-subtle card-title h5"
              >
                State Counter
              </div>
              <div>
                <div
                  class="_row_e43be3"
                >
                  <button
                    aria-label="Decrement value"
                    class="_button_e43be3"
                  >
                     - 
                  </button>
                  <span
                    aria-label="Counter value"
                    class="_value_e43be3"
                  >
                    0
                  </span>
                  <button
                    aria-label="Increment value"
                    class="_button_e43be3"
                  >
                     + 
                  </button>
                </div>
                <div
                  class="_row_e43be3"
                >
                  <input
                    aria-label="Set increment amount"
                    class="_textbox_e43be3"
                    value="2"
                  />
                  <button
                    aria-label="Add increment amount"
                    class="_button_e43be3"
                  >
                    Add
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
